fs-index/src/index.rs

```diff
@@ -474,4 +474,109 @@ impl<Id: ResourceId> ResourceIndex<Id> {
 
         Ok(IndexUpdate { added, removed })
     }
+
+    /// Track the addition of a new resource
+    ///
+    /// This method accepts a resource ID and returns `Result::Ok` if the
+    /// resource was successfully added to the index.
+    ///
+    /// **Note**: The called must ensure that:
+    /// - The index is up-to-date with the file system except for the new
+    ///   resource
+    /// - The resource is not already in the index
+    pub fn track_addition(&mut self, _id: Id) -> Result<()> {
```

Comment:
The use-case is an app that created resource, and it's very unlikely to have this resource indexed already (user created new image for instance). Then we can just add the resource to the index, skipping the whole index update.

So, the caller must ensure that:
1. The index is up-to-date except this single id.
2. The id hasn't been indexed before.

All `track` functions are intended to be used by app developers for optimized index manipulations, but they should understand the circumstances.

Reply:
We also can enforce invariant using `debug_assert!` that the id hasn't been indexed before.

Reply:
Now `update_one` makes sense to me again... 😅

Reply:
`update_one()` is quite similar to `track_modification()` in that both aim to update a resource, whether it was modified, added, or deleted. The main difference I see is that in this PR, `track_modification()` was designed to accept the resource ID instead of its path.

I support having a method like `update_one()` that checks the file system for changes and updates the index for just that specific resource. This would be useful in cases where the user doesn't want to run `update_all()`, which performs a full rescan.

The function definition should look something like this:
```rust
/// Update the index with the latest information from the file system
/// for a single resource
///
/// This method accepts the path of a single resource and updates the index
/// regardless of whether the resource was added, removed, or modified.
///
/// **Note**: The caller must ensure that:
/// - The index is up-to-date with the file system except for the updated
///  resource
/// - The resource was already in the index
pub fn update_one(&mut self, _path: &Path) -> Result<()>
```
We could also return a more informative type that provides details about the specific kind of update that occurred in the file system.